### PR TITLE
Fixes #20 UnhandledRejection error

### LINKED ISSUE #20: Unhandled Promise Error
The fetch method doesn't currently have a catch, as such if the server is offline or something blocks the request, the app won't render because of an unhandled rejection.

I will fix this by adding a catch method to the fetch request.

Documentation on Fetch:

> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

### PR DESCRIPTION
**Pull request by brandon-pereira**

### CODE REVIEW

**Comment by perrin4869** [comment]
@brandon-pereira Thanks for the contribution! Any chance you can add a test case for this addition?

**Comment by brandon-pereira** [comment]
Added unit tests

**Comment by perrin4869** [comment]
hm... this test passes even if I remove your proposed change to the `src`

**Comment by brandon-pereira** [comment]
@perrin4869 I just updated the tests to work on the catch.. I guess I should have paid more attention, my bad. Removing the catch in src will now cause unit test to fail.

**Comment by perrin4869** [comment]
All set! Thanks!

File: src/index.js [hunk 1]
```diff
@@ -76,7 +76,8 @@ class Backend {
         } catch (e) {
           return callback(`failed parsing ${url} to json`, false);
         }
-      });
+      })
+      .catch(() => callback(`failed loading ${url}`, false));
```
Comment (deleteme) [inline src/index.js]: This `catch` applies to the following scenarios:

1. The rejection of `fetch`.
2. A rejection or error in the first `then` call.
3. A rejection or error in the second `then` call.

However the callback message `failed loading for ${url}` applies to the first scenario.

**Pull request merged**